addons/sale/controllers/portal.py

```diff
@@ -193,10 +208,20 @@ def _get_payment_values(self, order_sudo, downpayment=False, **kwargs):
         logged_in = not request.env.user._is_public()
         partner_sudo = request.env.user.partner_id if logged_in else order_sudo.partner_id
         company = order_sudo.company_id
-        if downpayment:
-            amount = order_sudo._get_prepayment_required_amount()
+        if not link_amount:
+            if downpayment:
+                amount = order_sudo._get_prepayment_required_amount() - order_sudo.amount_paid
+            else:
+                amount = order_sudo.amount_total - order_sudo.amount_paid
         else:
-            amount = order_sudo.amount_total - order_sudo.amount_paid
+            if installment == 'false':
```

Comment:
You defined `installment`   as a boolean by default but here it is checked as a string. Is it intended? If yes, could you add a comment?

Reply:
Indeed, fixed.